FWCore/Framework/src/PathStatusInserter.cc

```diff
@@ -0,0 +1,27 @@
+
+#include "FWCore/Framework/src/PathStatusInserter.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Utilities/interface/StreamID.h"
+
+#include <memory>
+
+namespace edm
+{
+  PathStatusInserter::PathStatusInserter(unsigned int numberOfStreams) :
+    hltPathStatus_(numberOfStreams)
+  {
+    produces<HLTPathStatus>();
+  }
+
+  void
+  PathStatusInserter::setPathStatus(StreamID const& streamID,
+                                    HLTPathStatus const& hltPathStatus) {
+    hltPathStatus_[streamID.value()] = hltPathStatus;
```

Comment:
After discussion, we determined this is thread safe since the action of setting a value is done on the same thread just before the value will be read.